### PR TITLE
DMP-697: 'Your Audio' screen

### PR DESCRIPTION
**Pull request by deanbullock-hmcts**
### Jira link (if applicable)

[DMP-697](https://tools.hmcts.net/jira/browse/DMP-697)

### CODE REVIEW

File: src/app/components/hearing/hearing.component.html [hunk 1]
```diff
@@ -74,7 +74,7 @@ <h2 id="events-audio-heading" class="govuk-heading-m">Events and audio recording
       [audioRequest]="requestObject"
       [requestId]="requestId"
       [state]="state"
-      [userProfile]="userProfile$ | async"
```
[the remark this answers is not in the document]
Comment (deanbullock-hmcts) [reply src/app/components/hearing/hearing.component.html:77]: That's the deletion, see below :)

[the remark this answers is not in the document]
Comment (bpires2) [reply src/app/components/hearing/hearing.component.html:77]: ah ok

File: src/app/components/audios/audios.component.html [hunk 2]
```diff
@@ -1 +1,65 @@
-<h3 class="govuk-heading-s">Your Audio</h3>
+<ng-template #tableTemplatePartial let-row>
+  <td class="govuk-table__cell">
+    <a [routerLink]="['.']"> {{ row.caseNumber }}</a>
```
Comment (davet1985) [inline src/app/components/audios/audios.component.html:3]: Should link to case screen?

File: src/app/components/audios/audios.component.html [hunk 3]
```diff
@@ -1 +1,65 @@
...
+      <ng-container *ngIf="data.inProgessRows.length">
+        <h2 class="govuk-heading-m">In Progress</h2>
+        <app-data-table [rows]="data.inProgessRows" [columns]="columns" [pagination]="false">
+          <ng-template [tableRowTemplate] let-row>
+            <!-- Partial template for duplicated cells -->
+            <ng-container *ngTemplateOutlet="tableTemplatePartial; context: { $implicit: row }"></ng-container>
+
+            <td class="govuk-table__cell">
+              <strong *ngIf="row.status === 'FAILED'" class="govuk-tag govuk-tag--red"> {{ row.status }} </strong>
+              <strong *ngIf="row.status !== 'FAILED'" class="govuk-tag govuk-tag--yellow"> {{ row.status }} </strong>
```
Comment (davet1985) [inline src/app/components/audios/audios.component.html:32]: Might be cleaner using `ngClass`? (not a deal-breaker)

File: src/app/components/audios/audios.component.ts [hunk 4]
```diff
@@ -1,9 +1,114 @@
...
+      inProgessRows: this.inProgessRows$,
+      completedRows: this.completedRows$,
+      expiredRows: this.expiredRows$,
+    });
+  }
+
+  mapAudioRequestToRows(audioRequests: UserAudioRequest[]): UserAudioRequestRow[] {
+    return audioRequests.map((ar) => {
+      return {
+        // id: ar.case_id,
```
Comment (davet1985) [inline src/app/components/audios/audios.component.ts:89]: Might need this for linking to the case screen?

File: src/app/components/audios/audios.component.ts [hunk 5]
```diff
@@ -1,9 +1,114 @@
...
+        caseNumber: ar.case_number,
+        courthouse: ar.courthouse_name,
+        hearingDate: this.datePipe.transform(ar.hearing_date, 'dd MMM yyyy'),
+        startTime: this.datePipe.transform(ar.media_request_start_ts, 'hh:mm:ss', 'utc'),
+        endTime: this.datePipe.transform(ar.media_request_end_ts, 'hh:mm:ss', 'utc'),
+        requestId: ar.media_request_id,
+        expiry: this.datePipe.transform(ar.media_request_expiry_ts, 'hh:mm:ss dd/mm/yyyy', 'utc'),
```
Comment (davet1985) [inline src/app/components/audios/audios.component.ts:96]: Is it better to use the pipe here or in the template?

File: src/app/components/case/hearing-results/hearing-results.component.html [hunk 6]
```diff
@@ -1,12 +1,6 @@
...
-
-    <li class="moj-sub-navigation__item">
-      <a class="moj-sub-navigation__link" href="#2">Transcripts</a>
-    </li>
-  </ul>
-</nav>
-<app-data-table [rows]="rows" [columns]="columns"></app-data-table>
+<app-tabs>
+  <div *tab="'Hearings'">
+    <app-data-table [rows]="rows" [columns]="columns"></app-data-table>
```
Comment (davet1985) [inline src/app/components/case/hearing-results/hearing-results.component.html:3]: 🎉

File: src/app/components/hearing/hearing.component.html [hunk 7]
```diff
@@ -74,7 +74,7 @@ <h2 id="events-audio-heading" class="govuk-heading-m">Events and audio recording
...
+      [userProfile]="userState"
```
Comment (davet1985) [inline src/app/components/hearing/hearing.component.html:77]: Is it worth amending the name of the input to better match?

**Review by davet1985 — CHANGES_REQUESTED** [review]
A few minor comments, also

- Can we have a functional test to cover this please.
- Can you add to the pa11y tests please.

File: src/app/components/audios/audios.component.html (hunk 2 above)
Comment (deanbullock-hmcts) [reply src/app/components/audios/audios.component.html:3]: The current response doesn't include a case_id only a case_number therefore we are unable to link back currently. I did mention this too @hemantasharma1129 and he raised a ticket I think.

File: src/app/components/audios/audios.component.ts (hunk 4 above)
Comment (deanbullock-hmcts) [reply src/app/components/audios/audios.component.ts:89]: Again, awaiting the case_id to be added to the response.

File: src/app/components/audios/audios.component.html (hunk 2 above)
Comment (davet1985) [reply src/app/components/audios/audios.component.html:3]: Noted.

File: src/app/components/audios/audios.component.ts (hunk 5 above)
Comment (deanbullock-hmcts) [reply src/app/components/audios/audios.component.ts:96]: I did it this way initially as the table didn't support the dynamic templating / ability to add datepipe in the table template. Doing it here in the ts mapping allows me to do it once or potentially 3 times in the template. Having said that, I have a template partial that I reuse now so I don't think if it matters either way apart from preference or readability I guess.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**